python/plot_hist_sgp4Scan.py

```diff
@@ -0,0 +1,178 @@
+
+print "Fetching scan data from database ..."
+
+# Connect to SQLite database.
+try:
+    database = sqlite3.connect(config['database'])
+
+except sqlite3.Error, e:
+    print "Error %s:" % e.args[0]
+    sys.exit(1)
+
+# Fetch scan data.
+scan_data = pd.read_sql( "SELECT    " + config['error'] + "_x_error, \
+                                    " + config['error'] + "_y_error, \
+                                    " + config['error'] + "_z_error, \
+                                    " + config['error'] + "_error    \
+                          FROM      sgp4_scanner_results             \
+                          WHERE     success = 1;",
+                          database )
+
+scan_data.columns = [ 'xerror', 'yerror', 'zerror', 'magnitudeError' ]
+
+# The histogram of the data
+xerror = scan_data[ 'xerror' ]
+yerror = scan_data[ 'yerror' ]
+zerror = scan_data[ 'zerror' ]
+magnitudeError = scan_data[ 'magnitudeError' ]
+
+# Calculate mean, variance and standard deviation
+mu = sum( magnitudeError ) / len( magnitudeError )
+print 'Mean = ' + repr( mu )
+
+sumOfSquareDeviations = sum( ( x - mu )**2 for x in magnitudeError )
+variance = sumOfSquareDeviations / len( magnitudeError )
+print 'Variance = ' + repr( variance )
+
+sigma = variance**0.5
+print 'Standard Deviation = ' + repr( sigma )
+
+# Plot the magnitude of the error
+n, bins, patches = plt.hist( magnitudeError, bins=100, normed=True, facecolor='green', alpha=0.5,
+                             label='Magnitude' )
+
+# Add a line of expected distribution
+# y = mlab.normpdf( bins, mu, sigma )
+# l = plt.plot( bins, y, 'r--', linewidth=1.5, label='Best fit line' )
+
+# Select appropriate unit and title for the error type
+if config['error'] == 'arrival_position':
+    errorUnit = "[km]"
+    plotTitle = 'Arrival Position Error'
+else:
+    errorUnit = "[km/s]"
+    plotTitle = 'Arrival Velocity Error'
+
+# Figure properties
+plt.xlabel( 'Error' + " " + errorUnit )
+plt.ylabel( 'Normed Frequency' )
+plt.title( plotTitle + " " + 'Magnitude' )
+# plt.axis([40, 160, 0, 0.03])
+plt.legend( )
+plt.grid( True )
+
+# Save figure to file.
+plt.savefig( output_path_prefix + config["histogram_figure"] + "_" + config['error'] + "_error"
+             + "_magnitude" + ".png", dpi=config["figure_dpi"] )
+plt.close( )
+
+# Plot the components of the error in a separate figure.
+n, bins, patches = plt.hist( xerror, bins=200, histtype='step', normed=False, facecolor='red',
+                             alpha=1, label='X Axis' )
+n, bins, patches = plt.hist( yerror, bins=200, histtype='step', normed=False, facecolor='blue',
+                             alpha=1, label='Y Axis' )
+n, bins, patches = plt.hist( zerror, bins=200, histtype='step', normed=False, facecolor='orange',
+                             alpha=1, label='Z Axis' )
+# Figure properties
+plt.xlabel( 'Error' + " " + errorUnit )
+plt.ylabel( 'Frequency' )
+plt.title( plotTitle + " " + 'Components' )
```

Comment:
For papers we don't want to produce titles, maybe add a config option for this as well